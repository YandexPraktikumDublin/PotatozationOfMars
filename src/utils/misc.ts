import httpContext from 'express-http-context'
import Cookies from 'universal-cookie'

export const isServer = (): boolean =>
  !(
    typeof window !== 'undefined' &&
    window.document &&
    window.document.createElement
  )

export const getInlineCookiesFromHttpContext = () =>
  Object.entries(httpContext.get('cookies') ?? {}).reduce(
    (accumulator, [key, value]) => {
      accumulator += `${key}=${value}; `

      return accumulator
    },
    ''
  )

export const clearCookies = () => {
  const cookies = new Cookies()

  Object.entries(cookies.getAll() ?? {}).forEach(([key]) => {
    cookies.remove(key)
  })
}

export const hardRedirectTo = (location: string): void => {
  if (!isServer()) window.location.href = location
}

export const formatDate = (date?: string): string =>
  date
    ? new Date(date).toLocaleDateString('en-US', {
        year: 'numeric',
        month: 'numeric',
        day: 'numeric',
        hour: 'numeric',
        minute: 'numeric'
      })
    : ''

export const generatePassword = (): string =>
  Math.random().toString(36).slice(-8)

export const setStyleVariable = (name: string, value: string): void => {
  if (!isServer()) document.documentElement.style.setProperty(name, value)
}
