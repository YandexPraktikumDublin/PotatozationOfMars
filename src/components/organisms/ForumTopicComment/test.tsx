import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import { formatDate } from '@utils/misc'

import { ForumTopicComment } from './index'

describe('<ForumTopicComment />', () => {
  const comment = {
    id: 1,
    content: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
    userId: 1,
    user: {
      id: 1,
      login: 'IvanIvanov',
      name: 'Ivan Ivanov',
      createdAt: '01.12.20',
      updatedAt: '01.12.20'
    },
    topicId: 1,
    reactions: [
      {
        id: 1,
        content: 'ðŸ˜‚',
        userId: 1,
        user: {
          id: 1,
          login: 'IvanIvanov',
          name: 'Ivan Ivanov',
          createdAt: '01.12.20',
          updatedAt: '01.12.20'
        },
        commentId: 2,
        createdAt: '01.12.20',
        updatedAt: '01.12.20'
      }
    ],
    createdAt: '01.12.20',
    updatedAt: '01.12.20'
  }

  it('should renders correct <ForumTopicComment />', () => {
    const wrapper = shallow(<ForumTopicComment comment={comment} />)

    expect(wrapper.contains(comment.content)).toBeTruthy()
    expect(wrapper.contains(comment.user.name)).toBeTruthy()
    expect(wrapper.contains(formatDate(comment.createdAt))).toBeTruthy()

    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})
