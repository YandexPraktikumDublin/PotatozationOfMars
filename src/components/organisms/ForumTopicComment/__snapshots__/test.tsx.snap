// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForumTopicComment /> should renders correct <ForumTopicComment /> 1`] = `
<Fragment>
  <div
    className="text-left border border-primary rounded-2xl p-4 mb-4 last:mb-0 dark:border-white"
  >
    <div
      className="flex justify-between items-center mb-2 text-sm"
    >
      <span
        className=""
      >
        Ivan Ivanov
         at 
        1/12/2020, 12:00 AM
      </span>
      <button
        className="underline"
        onClick={[Function]}
        title="Add reaction"
      >
        Reply
      </button>
    </div>
    <div
      className="mb-3 text-lg"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit.
    </div>
    <ForumTopicCommentActions
      comment={
        Object {
          "content": "Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
          "createdAt": "01.12.20",
          "id": 1,
          "reactions": Array [
            Object {
              "commentId": 2,
              "content": "ðŸ˜‚",
              "createdAt": "01.12.20",
              "id": 1,
              "updatedAt": "01.12.20",
              "user": Object {
                "createdAt": "01.12.20",
                "id": 1,
                "login": "IvanIvanov",
                "name": "Ivan Ivanov",
                "updatedAt": "01.12.20",
              },
              "userId": 1,
            },
          ],
          "topicId": 1,
          "updatedAt": "01.12.20",
          "user": Object {
            "createdAt": "01.12.20",
            "id": 1,
            "login": "IvanIvanov",
            "name": "Ivan Ivanov",
            "updatedAt": "01.12.20",
          },
          "userId": 1,
        }
      }
    />
  </div>
</Fragment>
`;
